lib/elixir/lib/io/ansi/docs.ex

```diff
@@ -53,6 +55,37 @@ defmodule IO.ANSI.Docs do
     newline_after_block()
   end
 
+  @doc """
+  Prints documentation metadata (only `since` and `deprecated` for now).
+
+  See `default_options/0` for docs on the supported options.
+  """
+  @spec print_metadata(map, keyword) :: :ok
+  def print_metadata(metadata, options \\ []) when is_map(metadata) do
```

Comment:
Can we please move this function to IEx.Introspection? We only keep in this module functionality used by both IEx and Mix and Mix is not currently using any of the metadata formatting. :)

Reply:
Another advantage of moving this to introspection is what we can unify the code with https://github.com/elixir-lang/elixir/pull/7886/files#diff-9500c68886c2a36c242cdad92fe0f5e6R743. Otherwise, some of the logic you added, such as deprecated wrapping, only applies if ANSI is enabled.

Reply:
To be precise: I would just get rid of the `write_with_wrap` logic when moving this to introspection and have both behave the same (without wrapping).

Reply:
We already print the docs without wrapping if ANSI is not enabled since the code simply calls `IO.puts(doc)` in that case. Moving this to `IEx.Introspection` would also require duplicating the wrapping logic there or exporting it from `IO.ANSI.Docs`.

I wouldn't want to duplicate code. Would you be okay with exporting `write_with_wrap/4`?

Reply:
The code also relies on a new `:doc_metadata` option for the color, and other helpers in `IO.ANSI.Docs`. It wouldn't be clear how to configure the color if moved.

Reply:
@lackac I see. So let's keep this here but what if we also change this code to be able to handle when the color is disabled too? Basically, if color is not enabled, we don't add the color+reset. Would that be a fair compromise?

Reply:
It would :)